src/ranking/ranking.ts

```diff
@@ -85,20 +85,39 @@ function getScore(model: SpecQueryModel, rankingName: string, schema: Schema, op
   return score;
 }
 
-export function comparator(name: string, schema: Schema, opt: QueryConfig) {
+export function comparator(name: string | string[], schema: Schema, opt: QueryConfig) {
   return (m1: SpecQueryModel, m2: SpecQueryModel) => {
-    return getScore(m2, name, schema, opt).score - getScore(m1, name, schema, opt).score;
+    if (name instanceof Array) {
+      return getScoreDifference(name, m1, m2, schema, opt);
+    } else {
+      return getScore(m2, name, schema, opt).score - getScore(m1, name, schema, opt).score;
+    }
   };
 }
 
-export function groupComparator(name: string, schema: Schema, opt: QueryConfig) {
+export function groupComparator(name: string | string[], schema: Schema, opt: QueryConfig) {
   return (g1: SpecQueryModelGroup, g2: SpecQueryModelGroup) => {
     const m1 = g1.getTopSpecQueryModel();
     const m2 = g2.getTopSpecQueryModel();
-    return getScore(m2, name, schema, opt).score - getScore(m1, name, schema, opt).score;
+    if (name instanceof Array) {
+      return getScoreDifference(name, m1, m2, schema, opt);
+    } else {
+      return getScore(m2, name, schema, opt).score - getScore(m1, name, schema, opt).score;
+    }
   };
 }
 
+function getScoreDifference(name: string[], m1: SpecQueryModel, m2: SpecQueryModel, schema, opt): number {
+    let scoreDifference = getScore(m2, name[0], schema, opt).score - getScore(m1, name[0], schema, opt).score;
+    for (let i = 1; i < name.length; i++) {
```

Comment:
Why don't you start at 0 and just call `getScore` only once. 
Instead of break, you can just return things right away.  

And return 0 at the end.